core/modules/qproc/SecondaryIndex.cc

```diff
@@ -106,40 +95,106 @@ class MySqlBackend : public SecondaryIndex::Backend {
-                                              c.params[2], c.params[3]);
-        ChunkSpecMap m;
-        for(std::shared_ptr<sql::SqlResultIter> results
-                = _sqlConnection.getQueryIter(lookupSql);
-            !results->done();
-            ++(*results)) {
-            StringVector const& row = **results;
-            int chunkId = boost::lexical_cast<int>(row[0]);
-            int subChunkId = boost::lexical_cast<int>(row[1]);
-            ChunkSpecMap::iterator e = m.find(chunkId);
-            if (e == m.end()) {
-                ChunkSpec& cs = m[chunkId];
-                cs.chunkId = chunkId;
-                cs.subChunks.push_back(subChunkId);
-            } else {
-                ChunkSpec& cs = e->second;
-                cs.subChunks.push_back(subChunkId);
+
+private:
+
+
+    static std::string _buildIndexTableName(
+        std::string const& db,
+        std::string const& table
+    ) {
+        return (std::string(SEC_INDEX_DB) + "."
+                + sanitizeName(db) + "__" + sanitizeName(table));
+    }
+
+    /**
+     *  Build sql query string to run against secondary index
+     *
+     *  @param:    vector of string used to build the query,
+     *             format is:
+     *             [db, table, keyColumn, id_0, ..., id_n]
+     *             where:
+     *             - db.table is the director table,
+     *             - keyColumn is its primary key,
+     *             - id_x are keyColumn values
+     *
+     *  @return:   the sql query string to run against secondary index in
+     *             order to get (chunks, subchunks) couples containing [id_0, ..., id_n]
+     */
+    static std::string _buildLookupQuery(
+        std::vector<std::string> const& params
+    ) {
+
+        char const LOOKUP_SQL_TEMPLATE[] = "SELECT %s, %s FROM %s WHERE %s IN (%s);";
+        LOGF(getLogger(), LOG_LVL_TRACE, "params: %s" % util::formatable(params));
+
+        std::string const& db = params[0];
+        std::string const& table = params[1];
+        std::string const& keyColumn = params[2];
+
+        char const *const empty_bracket = "";
+        auto idsFormatter = util::formatable(params, 3, empty_bracket, empty_bracket);
+        std::string sql = (boost::format(LOOKUP_SQL_TEMPLATE) % CHUNK_COLUMN
+                                                              % SUB_CHUNK_COLUMN
+                                                              % _buildIndexTableName(db, table)
+                                                              % keyColumn
+                                                              % idsFormatter).str();
+        LOGF(getLogger(), LOG_LVL_TRACE, "sql: %s" % sql);
+        return sql;
+    }
+
+    /**
+     *  Add results from secondary index sql query to existing ChunkSpec vector
+     *
+     *  @param constraint:  a secondary index constraint issued from query analysis
+     *  @param output:      existing ChunkSpec vector
+     *  @result:            void
```

Comment:
I think @return is more frequently used tag. And you can skip it if it "returns" void.